tests/model_registry/conftest.py

```diff
@@ -467,3 +521,646 @@ def is_model_registry_oauth(request: FixtureRequest) -> bool:
+            deployment = Deployment(name=db_name, namespace=config.namespace, ensure_exists=True)
+        else:
+            # Create resources using individual factories
+            secret_config = SecretConfig(
+                name=db_name,
+                namespace=config.namespace,
+                string_data={
+                    "database-name": config.database_name,
+                    "database-user": config.database_user,
+                    "database-password": config.database_password,
+                },
+                labels=labels,
+                annotations=config.annotations or MODEL_REGISTRY_DB_SECRET_ANNOTATIONS,
+                teardown=config.teardown,
+            )
+            secret = secret_factory(secret_config)  # noqa: FCN001
+
+            pvc_config = PVCConfig(
+                name=db_name,
+                namespace=config.namespace,
+                access_modes=config.access_mode,
+                size=config.storage_size,
+                labels=labels,
+                teardown=config.teardown,
+            )
+            pvc = pvc_factory(pvc_config)  # noqa: FCN001
+
+            service_config = ServiceConfig(
+                name=db_name,
+                namespace=config.namespace,
+                ports=[
+                    {
+                        "name": "mysql",
+                        "nodePort": 0,
+                        "port": config.port,
+                        "protocol": "TCP",
+                        "appProtocol": "tcp",
+                        "targetPort": config.port,
+                    }
+                ],
+                selector={"name": db_name},
+                labels=labels,
+                annotations={
+                    "template.openshift.io/expose-uri": (
+                        "mysql://{.spec.clusterIP}:{.spec.ports[?(.name==\\mysql\\)].port}"
+                    ),
+                },
+                teardown=config.teardown,
+            )
+            service = service_factory(service_config)  # noqa: FCN001
+
+            deployment_config = DeploymentConfig(
+                name=db_name,
+                namespace=config.namespace,
+                template=get_model_registry_deployment_template_dict(secret_name=secret.name, resource_name=db_name),
+                labels=labels,
+                annotations={"template.alpha.openshift.io/wait-for-ready": "true"},
+                teardown=config.teardown,
+            )
+            deployment = deployment_factory(deployment_config)  # noqa: FCN001
+
+        bundle = ModelRegistryDBBundle(service=service, pvc=pvc, secret=secret, deployment=deployment, config=config)
+        created_bundles.append(bundle)
+
+        # Note: Cleanup is now handled by the individual resource factories.
+        # We don't need to register a cleanup for the bundle itself.
+
+        LOGGER.info(f"Successfully created DB bundle: {db_name}")
```

Comment:
This one is too much. I don't like this part. Can this be shortened and simplified?